index.html

```
@@ -767,19 +859,175 @@ <h2><dfn>CaptureController</dfn></h2>
+                    <td>
+                      Set to true when the focus decision is finalized.
+                    </td>
+                  </tr>
+                  <tr>
+                    <td><dfn>[[\FocusBehavior]]</dfn></td>
+                    <td><code>null</code></td>
+                    <td>
+                      The focus behavior desired by the application.
+                    </td>
+                  </tr>
+                </tbody>
+              </table>
+            </dd>
+            <dt>
+              <dfn>setFocusBehavior</dfn>
+            </dt>
             <dd>
               <p>
-                Create one internal slot: <dfn data-dfn-for="CaptureController">[[\isBound]]</dfn>,
-                initialized to <code>false</code>.
+                Run the following steps:
               </p>
+              <ol>
+                <li>
+                  <p>
+                    Let <var>focusBehavior</var> be the method's first argument.
+                  </p>
+                </li>
+                <li>
+                  <p>
+                    If {{CaptureController/[[Source]]}} is <code>null</code>,
+                    set {{CaptureController/[[FocusBehavior]]}} to <var>focusBehavior</var>
+                    and abort these steps.
+                  </p>
+                </li>
+                <li>
+                  <p>
+                    If {{CaptureController/[[Source]]}} has been
+                    <a data-cite="GETUSERMEDIA#source-stopped">stopped</a>,
+                    [=exception/throw=] an "{{InvalidStateError}}" {{DOMException}}.
+                  </p>
+                </li>
+                <li>
+                  <p>
+                    If {{CaptureController/[[DisplaySurfaceType]]}} is neither
+                    {{DisplayCaptureSurfaceType/"browser"}} nor {{DisplayCaptureSurfaceType/"window"}},
+                    [=exception/throw=] an "{{InvalidStateError}}" {{DOMException}}.
```

First thought: this seems like an error of type, not state. Maybe `TypeError` and move it up one step?

Second thought: the "type" varies by what the user chooses in the picker... should we abort silently instead? Seems a footgun otherwise: Developer tests their code without error, but forgot to manually test choosing "Entire desktop" in the prompt which will throw in production.

Also, "Entire desktop" is already "in focus' so to speak, so why not call this success instead of failure?

---

> Are we OK with that?

I've commented on that elsewhere. The comment threads are getting out of hand. I propose that we merge this and iterate. If there are substantive issues to resolve first, like methods vs. attribute, let's focus on them and leave uncontroversial spec-language nits for a follow-up.

---

> Second thought: the "type" varies by what the user chooses in the picker... should we abort silently instead?

We do abort silently if `setFocusBehavior()` was called before gDM, when the surface type was unknown. If the call is made after the user makes their choice, failing to check the surface type is a Web-dev bug and should be fixed. By failing silently we would be delaying the time when the Web-dev would realize their mistake.

---

> Also, "Entire desktop" is already "in focus' so to speak, so why not call this success instead of failure?

That depends on the OS and whether it's the current desktop or not.
At any rate, I really think this is all good enough for v2.